### DISCUSSION TITLE
Code is stucked after authentication

### DISCUSSION THREAD

**Discussion by brenosilva** [O365/python-o365#867] [Q&A]
Hello @janscas 

I´m starting testing O365 library using this small piece of code but it is stucked after authentication message and didn´t retrieve users information:

`from O365 import Account

credentials = ('APP_ID_HERE', 'SECRET_ID_HERE')

account = Account(credentials, auth_flow_type='credentials', tenant_id='TENANT_ID_HERE')
if account.authenticate():
    print('Authenticated!')
    directory = account.directory()
    for user in directory.get_users():
        print(user)`

user@Ubuntu2204:~/tenchisecurity/o365$ python3 auth.py
Authenticated!

I would appreciate any help.
Thanks

**Comment by alejcas** [comment]
better use this construct:


```python
if not account.is_authenticated:
    account.authenticate()
```

Also can u enable logs:

```python
import logging

o365_logger = logging.getLogger('O365')
o365_logger.setLevel(logging.DEBUG)
```
